vignettes/rhino-fluent.Rmd

```diff
@@ -0,0 +1,712 @@
+    filtered_deals <- reactive({
+      shiny.fluent::fluentSalesDeals |> filter(
+        is_closed | input$includeOpen
+      )
+    })
+    output$datatable <- renderUI({
+      DetailsList(items = filtered_deals(), columns = columns)
+    })
+  })
+}
+```
+The `Stack` element in the code allows for stacking multiple elements vertically or horizontally.
+
+The horizontal argument is set to TRUE to horizontally stack the two elements. The tokens argument is a list that specifies the padding and the gap between the two elements. In this case, the padding is set to 20 and the childrenGap is set to 10. This will add a padding of 20 pixels and a gap of 10 pixels between the other `span` of the `Stack` element.
+
+The span element has a class attribute set to `ms-depth-8`. This is a class provided by the Microsoft fluent UI that adds a shadow effect to the element, giving it a depth effect. The `uiOutput` element is used to render the `DetailsList` element that contains the filtered data.
+
+The additional style for the class `ms-depth-8` is added in the `app/style/main.scss` file. So it now is modified to -
+```
+// app/style/main.scss
+
+.ms-depth-4 {
+    display: flex; 
+    flex-wrap: wrap;
+    padding: 5px;
+}
+
+.ms-depth-8 {
+    height: 500px; 
+    overflow: auto; 
+    width: 50%; 
+    display: inline-block;
+}
+```
+
+The Sass code limits the height of the table to 500 pixels, width to 50% of the screen size and allows scrolling through the data. Similar to the previous usecase, we would again have to repeat the Sass building process with `rhino` as the file has been updated.
+
+```
+# in R console
+rhino::build_sass()
+```
+
+So, this is how the UI should look now -
+
+```{r, echo=FALSE}
+knitr::include_graphics("images/rhino_fluent_gif_2.gif")
+```
+Note that we now we have space beside the table for additional UI component(s).
+
+## Render a Bar Chart with logic in a helper function
+
+We will now add a bar chart that plots the deal amounts closed by the sales reps. For this, we will code a helper function and keep it inside the `app/logic` folder.
+
+### Helper function in the logic folder of the `rhino` project
+
+* Go to the `app/logic` folder and create a new `.R` file. Lets name it `barchart.R`.
+* We shall use the `gglplot2` and `plotly` packages (which we installed in the beginning) for this purpose. 
+
+Here is the code -
+
+```
+# app/logic/barchart.R
+
+box::use(
+  ggplot2[aes, element_text, geom_bar, ggplot, labs, theme, theme_light],
+)
+
+#' @export
+generate_barchart <- function(filtered_data) {
+  ggplot(filtered_data, aes(x = rep_name, y = deal_amount, fill = rep_name)) +
+    geom_bar(stat = "identity") +
+    labs(x = "Sales Rep Name", y = "Deal Amount") +
+    theme_light() +
+    theme(axis.text.x = element_text(angle = 60, hjust = 1))
+}
+```
+
+The above code block loads the `ggplot2` package and generates a barchart with names of the sales reps on the X axis and the number of deals on the Y axis.
```

Comment:
This is incorrect, Y axis is not the number of deals but the summarized amount of deals.